shared/teams/team/settings/retention/index.js

```diff
@@ -0,0 +1,187 @@
+// @flow
+import * as React from 'react'
+import {globalColors, globalMargins, globalStyles, isMobile, platformStyles} from '../../../../styles'
+import {Box, ClickableBox, Icon, ProgressIndicator, Text} from '../../../../common-adapters'
+import {type MenuItem} from '../../../../common-adapters/popup-menu'
+import {type RetentionPolicy, type _RetentionPolicy} from '../../../../constants/types/teams'
+
+export type Props = {
+  policy: RetentionPolicy,
+  teamPolicy?: RetentionPolicy,
+  onSelect: (policy: _RetentionPolicy, changed: boolean) => void,
+  isTeamWide: boolean,
+  onShowDropdown: (items: Array<MenuItem | 'Divider' | null>, target: ?Element) => void,
+}
+
+type State = {
+  selected: _RetentionPolicy,
+  items: Array<MenuItem | 'Divider' | null>,
+  showMenu: boolean,
+}
+
+const commonOptions = [1, 7, 30, 90, 365]
+
+class RetentionPicker extends React.Component<Props, State> {
+  state = {
+    selected: {type: 'retain', days: 0},
+    items: [],
+    showMenu: false,
+  }
+
+  _onSelect = (val: number | 'retain' | 'inherit') => {
+    let selected: _RetentionPolicy
+    if (typeof val === 'number') {
+      selected = {type: 'expire', days: val}
+    } else if (val === 'inherit') {
+      selected = {type: 'inherit', days: 0}
```

Comment:
the more i see this the more i think the type should be tighter. should just be {type: 'inherit'}

Reply:
talked it over elsewhere: it'd be nice to have this be a union of exact types but flow doesn't play nice with the record factory or instantiating them on the fly, as we need to do in this component. leaving it this way for now since anything that uses it should have to check `type` anyway.